webapp/Makefile

```diff
@@ -9,37 +9,83 @@ include $(TOPDIR)/Makefile.global
 # Subdirectories to recurse into for REC_TARGETS
 SUBDIRS = config
 
+maintainer-conf: .env.local
+
+# Run Symfony in dev mode (for maintainer-mode):
+.env.local:
+	@echo "Creating file '$@'..."
+	@echo "# This file was automatically created by 'make maintainer-conf' to run" > $@
+	@echo "# the DOMjudge Symfony application in developer mode. Adjust as needed." >> $@
+	@echo "APP_ENV=dev" >> $@
+
+# Dump autoload dependencies (including plugins)
+# This is needed since symfony/runtime is a Composer plugin that runs while dumping
+# the autoload file
+composer-dump-autoload:
+	composer $(subst 1,-q,$(QUIET)) dump-autoload -o -a
+
+composer-dump-autoload-dev:
+	composer $(subst 1,-q,$(QUIET)) dump-autoload
+
 copy-bundle-assets:
 # We can not use bin/console here, as when using a fakeroot,
 # the include paths are broken. We just copy in the data we need
 	-rm -rf public/bundles/nelmioapidoc
 	mkdir -p public/bundles/nelmioapidoc
-	cp -R ../lib/vendor/nelmio/api-doc-bundle/Resources/public/* public/bundles/nelmioapidoc/
+	cp -R ../webapp/vendor/nelmio/api-doc-bundle/Resources/public/* public/bundles/nelmioapidoc/
 
 clean-l:
 	-rm -rf public/bundles/nelmioapidoc
 
-install-domserver:
+# Change baseDir in composer autogenerated files
+define fix_composer_paths
+	for file in autoload_psr4.php autoload_classmap.php autoload_files.php autoload_namespaces.php ; do \
+		sed -i "s#^\$$baseDir = .*#\$$baseDir = dirname('$(domserver_webappdir)');#" $(1)/composer/$$file ; \
+	done
+	sed -i "s#__DIR__ \. '/\.\./\.\./\.\.' \. '/webapp#'$(domserver_webappdir)#" $(1)/composer/autoload_static.php
+endef
+
+install-domserver: composer-dump-autoload
 # This must be done first to install with the rest.
 	$(MAKE) copy-bundle-assets
 	$(INSTALL_DIR) $(DESTDIR)$(domserver_webappdir);
-	for d in bin config migrations public resources src templates tests ; do \
+	for d in bin config migrations public resources src templates tests vendor; do \
 		$(call install_tree,$(DESTDIR)$(domserver_webappdir),$$d) ; \
 	done
+	$(call fix_composer_paths,$(DESTDIR)$(domserver_webappdir)/vendor)
 # Change webapp/public/doc symlink
 	rm $(DESTDIR)$(domserver_webappdir)/public/doc
 	ln -s $(domjudge_docdir) $(DESTDIR)$(domserver_webappdir)/public/doc
 # Now change all relative symlinks in webapp/public to their correct paths
```

Comment:
Is this stil needed?

Reply:
I tested and it seems to work without, lets see if CI finds further issues.